database/certificate.go

```diff
@@ -703,8 +638,20 @@ SELECT
 	is_name_constrained,
 	permitted_names,
 	signature_algo,
-	raw_cert
-FROM certificates INNER JOIN (SELECT $1 UNION ALL SELECT issuer_id FROM parents) p(cert_id) ON p.cert_id=certificates.id`, tipCertID)
+	raw_cert`
+
+// GetAllCertsInChain returns the certificate chain of which endEntityID is the tip of,
+// which is to say it returns the certificate that signs endEntityID, and the
+// certificate that trusts that other certificate, etc.
+func (db *DB) GetAllCertsInChain(endEntityID int64) ([]certificate.Certificate, error) {
+	rows, err := db.Query(`
+WITH RECURSIVE parents AS (
+	SELECT ` + certRows + ` FROM trust WHERE cert_id=$1
```

Comment:
that doesn't make sense. you're selecting rows from the certificates table while querying the trust table.